Calculator/Calculator.cpp

```diff
@@ -197,6 +257,53 @@ string remove_spaces(string &old_str) {
     return new_str;                                 //Return new string
 }
 
+vector<string> make_sub_vec(vector<string> &equ_vec, int left_index, int right_index) {     //Return a vector that is a subset of original vector
+    vector<string> sub_vec;                                     //Initialize variable for sub vector
+    for (int i = left_index; i <= right_index; i++) {           //Iterate over indices from start position to end position
+        sub_vec.push_back(equ_vec[i]);                          //For each index, add that value to sub_vec
+    }
+    return sub_vec;                                             //Return sub_vec
+}
+
+void paren_calc(vector<string> &equ_vec) {                      //Resolve parentheses in equation
+    bool found_left_paren = false;                      //Initialize variables to track if parentheses have been found
+    bool found_right_paren = false;
+    do {
+        found_left_paren = found_right_paren = false;       //Reset tracker variables to false
+        vector<string> sub_vec;                             //Initialize variable to hold sub vector
+        int left_paren_index = 0;                           //Initialize variable to track index of last found open paren
+        for (int i = 0; i < equ_vec.size(); i++) {          //Iterate over each entry in the variable
+            if (equ_vec[i] == "(") {                //If the entry is an open paren, update index and tracker variables
+                left_paren_index = i;
+                found_left_paren = true;
+            }
+            else if (equ_vec[i] == ")") {           //If the entry is a close paren
+                if (found_left_paren == true) {     //Check if we have found a corresponding open paren
+                    found_right_paren = true;       //If so, update right_paren tracker
+                    if (left_paren_index + 1 == i) {        //If parentheses are empty - have nothing between them
+                        replace_sub_equ(equ_vec, left_paren_index, 2, "0"); //replace them with a zero
+                    }
+                    else {                              //If parentheses are not empty
+                        sub_vec = make_sub_vec(equ_vec, left_paren_index + 1, i - 1);   //Make sub-vector from the contents of parentheses
+                        exp_calc(sub_vec);          //Resolve the sub-vector
+                        replace_sub_equ(equ_vec, left_paren_index, i - left_paren_index + 1, sub_vec[0]);   //Replace the parentheses and their contents with the result of their contents
+                        vec_print(equ_vec);
+                        break;          //End for loop to restart checking from left side
+                    }
+                }

```

Comment:
once we get this much indentation it's also a good sign we should try to refactor some of this logic out into another function is possible